Documentation/apply official theme

lucieleonard opened the pull request:
**Component**: docs, build

**Context**:

We want to use the official Scality theme for MetalK8s docs.

**Summary**:

Apply the official Scality documentation theme everywhere, with small specificities depending on the destination of the build:

- for ReadTheDocs, we make the "home link" (used for the header Scality logo) and the back arrow point to `https://metal-k8s.readthedocs.io`
- for Scality-hosted docs, we make these links respectively `https://documentation.scality.com` and `https://documentation.scality.com/metalk8s`
- for embedded docs, we disable these links (and hide the back arrow) for now

We also add:

- a `release` tag to simulate a build from the latest Git tag
- a `scality-product` tag, to generate docs to be hosted on `documentation.scality.com`

You can use `O='-t release -t scality-product' tox -e docs` to pass these tags to Sphinx.

bert-e commented:
# Hello lucieleonard,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.5/documentation/apply-official-theme` with contents from `documentation/apply-official-theme`
and `development/2.5`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.5/documentation/apply-official-theme origin/development/2.5
 $ git merge origin/documentation/apply-official-theme
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.5/documentation/apply-official-theme
```

NicolasT commented:
/wait

NicolasT commented on VERSION:
```diff
@@ -1,4 +1,4 @@
 VERSION_MINOR=4
 VERSION_PATCH=2
-VERSION_SUFFIX=-dev
+VERSION_SUFFIX=
```
The base of this branch ain't right: it seems to be based on the tag. So, will need to wait for 2.4.2 to be released, then merged (@gdemonet), then reparent this change on `development/2.4`.

gdemonet replied:
Reparented on `development/2.4` after 2.4.4 tag was merged.

NicolasT commented on docs/conf.py:
```diff
@@ -71,14 +74,30 @@
 html_theme_options = {
     'logo_only': True,
+        'social_links': [
+        ("linkedin", "https://www.linkedin.com/company/scality/"),
```
Indenting doesn't seem right.

lucieleonard replied:
Fixed

NicolasT commented:
Will need to be reparented.

Also, commit message is not according to standards (missing empty line between heading and body).

NicolasT reviewed (changes requested):
Will need to be reparented.

Also, commit message not according to standards (missing empty line between header and body).

gdemonet committed:
docs: Use "scality_sphinx" theme

We want to use the custom Scality theme for our documentation included
in the releases (not ReadTheDocs) and the product ISO.
Note that we need to adjust the `pip-compile` tox environment to allow
the installation of this theme directly from GitHub (installing from a
repository doesn't work with the `--generate-hashes` option).

NicolasT commented on artwork/metalk8s-logo-vertical.svg:
```diff
@@ -1,56 +1,14 @@
-<?xml version="1.0" encoding="utf-8"?>
```
What happened here? The end result is not even a valid SVG file anymore?

gdemonet replied:
I edited in some web-based tool, will look into it. What's making you say it's not valid SVG?

NicolasT replied:
It's not even an XML file anymore if I read the diff right.

Also, why edit in whatever tool? As in, what needed to be changed? Our logo has been like this since... forever.

gdemonet replied:
Reverted

NicolasT commented on docs/index.rst:
```diff
@@ -15,19 +15,27 @@ Welcome to MetalK8s's documentation!
-==================
+      * :ref:`genindex`
+      * :ref:`modindex`
+      * :ref:`search`
```
Do we don't include the search etc. on a release build? Kinda odd.

gdemonet replied:
I think this was a request from @lucieleonard. In the new theme, we always show the Search bar in the header. For other topics, I don't think we show them at all.

It's easy to add back if you think we need to.

gdemonet replied:
Added back (and removed `modindex` as we don't have any auto-documented modules).

NicolasT commented on docs/conf.py:
```diff
@@ -65,20 +68,38 @@
+        ("facebook", "https://www.facebook.com/scality/"),
+    ],
+    'footer_links': [
+        ("Support", "https://support.scality.com"),
```
It'd be nice for this link to be to Github for open-source builds.

gdemonet replied:
Done

NicolasT commented on docs/conf.py:
```diff
@@ -65,20 +68,38 @@
+    'footer_links': [
+        ("Support", "https://support.scality.com"),
+        ("Knowledge Base", "https://support.scality.com/hc/en-us"),
+        ("Training", "https://training.scality.com"),
```
These don't apply in open-source builds.

gdemonet replied:
Removed for builds without the `scality-product` build tag.

gdemonet committed:
docs: Omit developer docs from released builds

gdemonet committed:
docs/index: Remove 'module index'

We don't have any autodocumented module to show, so this link was always
broken.

gdemonet commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **wait**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `documentation/apply-official-theme` into
`development/2.4`
* [w/2.5/documentation/apply-official-theme](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/documentation/apply-official-theme) will be merged into `development/2.5` 
* [w/2.6/documentation/apply-official-theme](https://github.com/scality/metalk8s/compare/development/2.6...w/2.6/documentation/apply-official-theme) will be merged into `development/2.6` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @tcarmet

* @thomasdanan

* @sayf-eddine-scality

* @lucieleonard

* @monpote

* @carlito-scality

* @cmonfort

* @nicolast

* @slaperche-scality

* @teddyandrieux

* @nootal

* @gdemonet

* @alexandre-allard-scality





The following reviewers are expecting changes from the author, or must review again:

* @nicolast

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @tcarmet

* @thomasdanan

* @sayf-eddine-scality

* @lucieleonard

* @monpote

* @carlito-scality

* @cmonfort

* @nicolast

* @slaperche-scality

* @teddyandrieux

* @nootal

* @gdemonet

* @alexandre-allard-scality

bert-e commented:
# History mismatch



Merge commit #cba7bdb65bb62e267c49667c8fffbeb1c20af975 on the integration branch
`w/2.5/documentation/apply-official-theme` is merging a branch which is neither the current
branch `documentation/apply-official-theme` nor the development branch
`development/2.5`.

It is likely due to a rebase of the branch `documentation/apply-official-theme` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

gdemonet commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `documentation/apply-official-theme` into
`development/2.4`
* [w/2.5/documentation/apply-official-theme](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/documentation/apply-official-theme) will be merged into `development/2.5` 
* [w/2.6/documentation/apply-official-theme](https://github.com/scality/metalk8s/compare/development/2.6...w/2.6/documentation/apply-official-theme) will be merged into `development/2.6` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @tcarmet

* @thomasdanan

* @sayf-eddine-scality

* @lucieleonard

* @monpote

* @carlito-scality

* @cmonfort

* @nicolast

* @slaperche-scality

* @teddyandrieux

* @nootal

* @gdemonet

* @alexandre-allard-scality

bert-e commented:
# History mismatch



Merge commit #c4acb25a7c52e2fed725957471117d77a728491f on the integration branch
`w/2.5/documentation/apply-official-theme` is merging a branch which is neither the current
branch `documentation/apply-official-theme` nor the development branch
`development/2.5`.

It is likely due to a rebase of the branch `documentation/apply-official-theme` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

gdemonet committed:
build/docs: Pass READTHEDOCS environment with tox

It was not done, hence not tested in CI.

gdemonet commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `documentation/apply-official-theme` into
`development/2.4`
* [w/2.5/documentation/apply-official-theme](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/documentation/apply-official-theme) will be merged into `development/2.5` 
* [w/2.6/documentation/apply-official-theme](https://github.com/scality/metalk8s/compare/development/2.6...w/2.6/documentation/apply-official-theme) will be merged into `development/2.6` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @tcarmet

* @thomasdanan

* @sayf-eddine-scality

* @lucieleonard

* @monpote

* @carlito-scality

* @cmonfort

* @nicolast

* @slaperche-scality

* @teddyandrieux

* @nootal

* @gdemonet

* @alexandre-allard-scality

gdemonet committed:
docs: Add PyYAML to requirements

Since we parse salt/defaults.yaml to infer some tags, we need this
dependency to be installed.

gdemonet committed:
docs: Ensure copyright date is uptodate

Let's make it rely on the build-time information.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @chengyanjin

* @slaperche-scality

* @thomasdanan

* @lucieleonard

* @gdemonet

* @teddyandrieux

* @nootal

* @monpote

* @tcarmet

* @sayf-eddine-scality

* @cmonfort

* @nicolast

* @carlito-scality

* @alexandre-allard-scality

gdemonet commented:
/approve

bert-e commented:
# Not Author



I'm afraid I cannot do that, @gdemonet:

Only the author of the pull request @lucieleonard can use this command.

Please **delete** the corresponding comment so I can move on.

bert-e commented:
# Not Author



I'm afraid I cannot do that, @gdemonet:

Only the author of the pull request @lucieleonard can use this command.

Please **delete** the corresponding comment so I can move on.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @chengyanjin

* @slaperche-scality

* @thomasdanan

* @lucieleonard

* @gdemonet

* @teddyandrieux

* @nootal

* @monpote

* @tcarmet

* @sayf-eddine-scality

* @cmonfort

* @nicolast

* @carlito-scality

* @alexandre-allard-scality

gdemonet commented:
/bypass_author_approval

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.7`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **bypass_author_approval**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.7`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`



Please check the status of the associated issue None.

Goodbye lucieleonard.
